dotcms-observability/src/main/java/com/dotcms/observability/subsystem/AbstractSubsystem.java

```diff
@@ -0,0 +1,94 @@
+package com.dotcms.observability.subsystem;
+
+import com.dotcms.observability.resilence.ResilienceExecutor;
+import com.dotcms.observability.service.SystemStateService;
+import com.dotcms.observability.state.State;
+import com.dotcms.observability.state.StateChangeEvent;
+import jakarta.enterprise.event.Event;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+
+public abstract class AbstractSubsystem implements Subsystem {
```

Comment:
IMO abstract classes end up in smell code
I would prefer an interface and a base class (or composite pattern)